stdlib/Sockets/src/addrinfo.jl

```diff
@@ -319,3 +328,29 @@ function getipaddrs(addr_type::Type{T}=IPAddr; loopback::Bool=false) where T<:IP
     ccall(:uv_free_interface_addresses, Cvoid, (Ptr{UInt8}, Int32), addr, count)
     return addresses
 end
+
+"""
+    islinklocaladdr(addr::IPAddr)
+
+Tests if an IP address is a link-local address. Link-local addresses
+are not guaranteed to be unique beyond their network segment,
+therefore routers do not forward them. Link-local addresses are from
+the address blocks `169.254.0.0/16` or `fe80::/10`.
+
+# Example
+```julia
+filter(!islinklocaladdr, getipaddrs())
+```
+"""
+function islinklocaladdr(addr::IPv4)
+    # RFC 3927
+    return addr.host &
```

Comment:
This could use parens around the & I think. Looks odd otherwise.

Reply:
OK, parentheses now added in rebased commit b62e10c.